### PR TITLE
feat: add syncBalances

### PR DESCRIPTION
**Pull request by Cphilo**

### CODE REVIEW

**Review by merlincdj — APPROVED** [review]

File: contracts/WooStakingManager.sol [hunk 1]
```diff
@@ -260,6 +260,24 @@ contract WooStakingManager is IWooStakingManager, BaseAdminOperation, Reentrancy
         emit CompoundRewardsOnStakingManager(_user);
     }
 
+    function syncBalances(address[] memory _users, uint256[] memory _balances) external onlyAdmin {
+        require(_users.length == _balances.length, "WooStakingManager: INVALID_INPUTS");
+        uint256 len = _users.length;
+        address user;
+        uint256 balance;
+        uint256 prevBalance;
+        for (uint256 i = 0; i < len; ++i) {
```
Comment (fb-alexcq) [inline contracts/WooStakingManager.sol:269]: 加一个 unchecked { } ，用来节省gas吧。

Comment (Cphilo) [reply contracts/WooStakingManager.sol]: Resolved

File: contracts/WooStakingManager.sol [hunk 2]
```diff
@@ -260,6 +260,26 @@ contract WooStakingManager is IWooStakingManager, BaseAdminOperation, Reentrancy
         emit CompoundRewardsOnStakingManager(_user);
     }
 
+    function syncBalances(address[] memory _users, uint256[] memory _balances) external onlyAdmin {
+        require(_users.length == _balances.length, "WooStakingManager: INVALID_INPUTS");
+        uint256 len = _users.length;
+        address user;
+        uint256 balance;
+        uint256 prevBalance;
+        unchecked {
```
Comment (fb-alexcq) [inline contracts/WooStakingManager.sol:269]: for this unchecked behavior:
```
pragma solidity ^0.8.0;

import "hardhat/console.sol";

contract Test {
    constructor() {
    }

    function test1(uint256 prevBalance, uint256 balance) public {
        address user = 0xCD8C0FF0ADFA54b0Fa09e8409Cf5b598BdDFc76D;
        unchecked {
            uint256 len = 1;
            for (uint256 i = 0; i < len; ++i) {
                if (balance > prevBalance) {
                    stakeWoo(user, balance - prevBalance);
                } else if (balance < prevBalance) {
                    unstakeWoo(user, prevBalance - balance);
                }
            }
        }
    }

    function stakeWoo(address user, uint256 amount) public {
        console.log("In stakeWoo: ", user, amount);
        uint256 x = 5;
        uint256 y = 6;
        uint256 z = x - y;
        console.log(z);
    }   

    function unstakeWoo(address user, uint256 amount) public {
        console.log("In unstakeWoo: ", user, amount);
        unchecked {
            uint256 x = 5;
            uint256 y = 6;
            uint256 z = x - y;
            console.log(z);
        }
    }   
}
```
It works as we discussed online.

**Review by fb-alexcq — APPROVED** [review]

**Comment by fb-alexcq** [comment]
LGTM

**Pull request merged**